tests/bptree_map.rs

```diff
@@ -0,0 +1,80 @@
+use std::collections::{BTreeMap, BTreeSet};
+use std::ops::Bound;
+
+use concread::bptree::BptreeMap;
+
+proptest::proptest! {
+    #[test]
+    fn bptree_range_iter_consistent(values: BTreeSet<u8>, left in 0..u8::MAX - 1, len in 1..u8::MAX, bounds: (Bound<()>, Bound<()>)) {
+        let range = (bounds.0.map(|()| left), bounds.1.map(|()| left.saturating_add(len)));
+        let btree_map = BTreeMap::from_iter(values.iter().cloned().map(|v| (v, ())));
+        let bptree_map = BptreeMap::from_iter(values.iter().cloned().map(|v| (v, ())));
+        let bptree_map_read_tx = bptree_map.read();
+
+        let btree_iter = btree_map.range(range);
+        let bptree_iter = bptree_map_read_tx.range(range);
+
+        assert!(
+            btree_iter.eq(bptree_iter)
+        )
+    }
+
+    #[test]
+    fn bptree_get_consistent(values: BTreeSet<u8>, key: u8) {
+        let btree_map = BTreeMap::from_iter(values.iter().cloned().map(|v| (v, v)));
+        let bptree_map = BptreeMap::from_iter(values.iter().cloned().map(|v| (v, v)));
+        let bptree_map_read_tx = bptree_map.read();
+
+        let btree_value = btree_map.get(&key);
+        let bptree_value = bptree_map_read_tx.get(&key);
+
+        assert_eq!(btree_value, bptree_value);
+    }
+
+    #[test]
+    fn bptree_remove_consistent(values in proptest::collection::btree_set(proptest::arbitrary::any::<u8>(), 1..256), indices: Vec<proptest::sample::Index> ) {
+        let mut btree_map = BTreeMap::from_iter(values.iter().cloned().map(|v| (v.to_string(), v.to_string())));
```

Comment:
Minor nitpick, can you remove all the `to_string`'s here and in the next test - it's an unnecessary call and I couldn't work out why they're there.

Reply:
`to_strings` were necessary there to spot the issue, because strings require allocation and so we can spot double free

Reply:
`bptree_map::bptree_remove_consistent` doesn't fail before the patch in `src/internals/bptree/node.rs`...  `bptree_remove_1` does however.

Reply:
i've just tried to create `Box<u8>` but this way i can't reproduce the issue :(

Reply:
> bptree_map::bptree_remove_consistent doesn't fail before or after the patch 

interesting, on my machine `bptree_remove_consistent` fails every single time with errors like: 

```
lib-1f4d753d0d073d17(93725,0x16f473000) malloc: Double free of object 0x137708cd0
lib-1f4d753d0d073d17(93725,0x16f473000) malloc: *** set a breakpoint in malloc_error_break to debug
```

```
lib-1f4d753d0d073d17(93197,0x16f987000) malloc: Corruption of free object 0x1447079c0: msizes 37632/0 disagree
lib-1f4d753d0d073d17(93197,0x16f987000) malloc: *** set a breakpoint in malloc_error_break to debug
```

Reply:
Curious, what machine/os etc?

Reply:
Actually it fails now on rust v1.79, so I'm happy with this as-is.